src/App.js

```diff
@@ -30,8 +30,17 @@ class App extends Component {
         <Header />
         <Switch>
           <Route exact path="/" component={Home} />
-          <Route path="/dogindex" component={DogIndex} />
-          <Route path="/dogshow" component={DogShow} />
+          <Route
+            path="/dogindex"
+            render={(props) => <DogIndex dogs={this.state.dogs} />} />
+          <Route
+            path="/dogshow/:id"
+            render={(props) => {
+              let id = props.match.params.id
+              let dog = this.state.dogs.find(c => c.id === +id)
```

Comment:
I would consider changing the `c` in the find method to something that matches your dog naming conventions.